internal/config/config.go

```diff
@@ -319,6 +447,75 @@ func parsePeer(p peer) (*Peer, error) {
 	}, nil
 }
 
+// parsePeerAutodiscovery parses peer autodiscovery configuration, constructs
+// a PeerAutodiscovery and returns a pointer to it.
+func parsePeerAutodiscovery(p peerAutodiscovery) (*PeerAutodiscovery, error) {
```

Comment:
I find the presence of peerAutodiscovery -> the config and PeerAutodiscovery -> the object we use a bit confusing.
Maybe we can call peerAutodiscovery -> peerAutodiscoveryConfig?

Reply:
This is an existing pattern throughout the codebase at the moment. For example, we also have `peer` and `Peer`.

Reply:
Yep, and I agree consistency comes first.
Maybe we can amend later, I still find this a bit misleading, because both types belong to the same package but the only difference is that one is exported and the other is not.

Reply:
Fair enough @fedepaol. Please feel free to open a new issue to track this.

Reply:
@johananl I am a bit more familiar with the codebase now, so I understand the choice better.
It wasn't clear by reviewing the PR itself, because I was lacking context.
I am fine with leaving it as it is, or filing the issue, as you prefer :-) .